src/aktualizr_secondary/update_agent_file.cc

```diff
@@ -0,0 +1,62 @@
+#include "update_agent_file.h"
+#include "logging/logging.h"
+#include "uptane/manifest.h"
+
+// TODO: this is an initial version of a file update on Secondary
+bool FileUpdateAgent::isTargetSupported(const Uptane::Target& target) const { return target.type() != "OSTREE"; }
+
+bool FileUpdateAgent::getInstalledImageInfo(Uptane::InstalledImageInfo& installed_image_info) const {
+  installed_image_info.name = _target_name;
+  // TODO: proper verification, file exists, file open/read errors, etc
+  if (!_target_filepath.empty()) {
+    auto file_content = Utils::readFile(_target_filepath);
+    installed_image_info.len = file_content.size();
+    installed_image_info.hash = Uptane::ManifestIssuer::generateVersionHashStr(file_content);
+  } else {
+    // TODO: fake package manager
+    const std::string fake_pacman_case = "fake_pacman";
+    installed_image_info.len = fake_pacman_case.size();
+    installed_image_info.hash = Uptane::ManifestIssuer::generateVersionHashStr(fake_pacman_case);
```

Comment:
I'd probably prefer to leave the hash empty, although I'm open to discussion on what the right thing to do is. Either way, not enough to hold this up for now.

Reply:
The backend will reject a manifest with an empty hash value.